run_bin.go

```diff
@@ -27,7 +27,11 @@ type CoverageCollector struct {
 	coverMode              string
 	tmpCoverageFiles       []*os.File
 	setupFinished          bool
+	preCmdFuncs            []CmdFunc
+	postCmdFuncs           []CmdFunc
 }
+type CoverageCollectorOption func(collector *CoverageCollector)
+type CmdFunc func(cmd *exec.Cmd) error
```

Comment:
Do you think there should be a pre and post cmd func type? The pre command would be `func(cmd *exec.Cmd) error` and post commnd type would be `func(cmd *exec.Cmd, output string/[]byte?, error) error`.